common/flatpak-dir.c

```diff
@@ -4814,8 +4826,18 @@ flatpak_dir_pull_extra_data (FlatpakDir          *self,
           g_autoptr(GError) my_error = NULL;
 
           if (!g_file_load_contents (extra_local_file, cancellable, &extra_local_contents, &extra_local_size, NULL, &my_error))
-            return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Failed to load local extra-data %s: %s"),
-                                       flatpak_file_get_path_cached (extra_local_file), my_error->message);
+            {
+              if  (g_error_matches (my_error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
+                {
+                  g_propagate_prefixed_error (error, my_error, _("Failed to load local extra-data %s: %s"),
```

Comment:
If `my_error->message` is `"foo bar"`, this will return an error with message `"Failed to load local extra-data /path/to/extra_local_file: foo barfoo bar"`.